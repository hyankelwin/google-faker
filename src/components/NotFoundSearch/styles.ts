import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 20px;

  p {
    margin-top: 10px;

    span {
      margin-right: 5px;
      font-weight: 500;
    }
  }
`;
