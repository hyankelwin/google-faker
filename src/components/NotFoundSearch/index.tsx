import React from 'react';
import AnimalsConstant from '../../constants/animals.constant';
import { Container } from './styles';

interface NotFoundSearchProps {
  value: string;
}

const NotFoundSearch: React.FC<NotFoundSearchProps> = ({ value }) => {
  const handleRenderAnimals = () => {
    return (
      <>
        {AnimalsConstant.map(item => (
          <span key={item}>{item}, </span>
        ))}
      </>
    );
  };

  return (
    <Container>
      {value && (
        <p>
          No results found for <b>{`'${value}'.`}</b>
        </p>
      )}
      <p>Try looking for: {handleRenderAnimals()}</p>
    </Container>
  );
};

export default NotFoundSearch;
