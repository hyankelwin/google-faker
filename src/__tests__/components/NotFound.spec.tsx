import { mount } from 'enzyme';
import NotFoundSearch from '../../components/NotFoundSearch';

describe('NotFoundSearch component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render NotFoundSearch component with success', () => {
      const props = {
        value: 'dog',
      };

      const wrapper = mount(<NotFoundSearch {...props} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
