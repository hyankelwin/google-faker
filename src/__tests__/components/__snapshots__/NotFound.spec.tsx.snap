// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundSearch component should be able to passed in all tests on index.tsx should be able to render NotFoundSearch component with success 1`] = `
"<NotFoundSearch value=\\"dog\\">
  <styled.div>
    <div className=\\"sc-aXZVg iOvnNW\\">
      <p>
        No results found for 
        <b>
          &#39;dog&#39;.
        </b>
      </p>
      <p>
        Try looking for: 
        <span>
          bear
          , 
        </span>
        <span>
          bird
          , 
        </span>
        <span>
          cat
          , 
        </span>
        <span>
          cetacean
          , 
        </span>
        <span>
          cow
          , 
        </span>
        <span>
          crocodilia
          , 
        </span>
        <span>
          dog
          , 
        </span>
        <span>
          fish
          , 
        </span>
        <span>
          horse
          , 
        </span>
        <span>
          insect
          , 
        </span>
        <span>
          lion
          , 
        </span>
        <span>
          rabbit
          , 
        </span>
        <span>
          rodent
          , 
        </span>
        <span>
          snake
          , 
        </span>
      </p>
    </div>
  </styled.div>
</NotFoundSearch>"
`;
